### PR TITLE
feat(web): Add onboarding tutorial event

### LINKED ISSUE #112: First load of itinerary should have example tutorial event
### Expected Behavior

When user first starts an itinerary, an example event should be added.  This could have information in description like a small explanation of what you can do.

e.g: This is an example event!  You can add or change descriptions and titles by clicking the edit button at the side.  + search help when it's implemented, etc. etc.

### Current Behavior

Empty itinerary on first load

### PR DESCRIPTION
**Pull request by eccx400**
## Description
Launches the website with a Tutorial Event. Uses addTutorial Function from the PlanHelper class and is implemented in the componentDidMount() in Plan.tsx. As of them moment object is pushed on the array; however, the event is not visible on website. Tested setting addTutorial() function to a button and the object is created. Working on generating on initialization.

Changes to the Title seem to not revert to original title; something fishy with restoredData & restoredPlan, will look into it

## Tested
Console.log to see if the EventData Object is pushed onto the array and checks the length of the array (should be 1 on startup). Use yarn run web:dev to see changes on web.

## Related issues

* closes #112

### CODE REVIEW

File: packages/web/src/components/Plan.tsx [hunk 1]
```diff
@@ -105,23 +79,57 @@ export class Plan extends React.Component<{}, State> {
...
         super(props);
+
+        const theData = new EventData(
+            1,
+            "Tutorial Event",
+            new Date(),
+            new Date(),
+            "This is an example event! You can add or change descriptions and titles by clicking the edit button at the side. + search help when it's implemented, etc. etc."
+        );
+        this.state.events.push(theData);
```
Comment (drklee3) [inline packages/web/src/components/Plan.tsx]: Shouldn't modify the plan or state data here as it's updated in `componentDidMount()`, anything here will be overwritten.  The order of execution is `constructor()`, then `componentDidMount()` when it's first loaded.  Could also try moving the data restore + tutorial event logic here it prevents an extra render.

Tutorial will always show up for a brief second before the saved plan is loaded which isn't ideal.

File: packages/web/src/components/Plan.tsx [hunk 2]
```diff
@@ -105,23 +79,57 @@ export class Plan extends React.Component<{}, State> {
...
+        console.log("The EventData name is " + theData.name);
+        console.log("The EventData description is " + theData.description);
+        console.log("The Elements in the Array is " + this.state.events);
+        console.log("The No. of Elements is " + this.state.events.length);
     }
 
     componentDidMount(): void {
         const restoredData = restoreData();
         const restoredPlan = restorePlan();
+        const newData = new EventData(
```
Comment (drklee3) [inline packages/web/src/components/Plan.tsx]: nit: could use something like `tutorialEvent` as a variable name to be more clear

File: packages/web/src/components/Plan.tsx [hunk 3]
```diff
@@ -3,38 +3,12 @@ import { Button, Card, Col, Empty, Icon, Row, Tooltip, Typography } from "antd";
...
-import {
-    DragDropContext,
-    Draggable,
-    DraggableProvided,
-    Droppable,
-    DroppableProvided,
-    DropResult,
-    ResponderProvided,
-} from "react-beautiful-dnd";
+import { DragDropContext, Draggable, DraggableProvided, Droppable, DroppableProvided, DropResult, ResponderProvided } from "react-beautiful-dnd";
```
Comment (drklee3) [inline packages/web/src/components/Plan.tsx]: Were these auto formatted btw?  vscode, assuming you have the `prettier` extension installed should keep it formatted like before this change.

File: packages/web/src/components/Plan.tsx [hunk 4]
```diff
@@ -105,23 +79,57 @@ export class Plan extends React.Component<{}, State> {
...
         }
 
-        // if no restored state, use an empty plan and default title
+        // if no restored state, revert to launch tutorial
         this.setState({
             data: restoredData
                 ? { title: restoredData.title }
                 : { title: "Your Plan" },
-            events: restoredPlan ? restoredPlan : [],
+            events: restoredPlan ? restoredPlan : arr,
```
Comment (drklee3) [inline packages/web/src/components/Plan.tsx:133]: This should be the correct way to add the tutorial event :+1:

**Review by drklee3 — COMMENTED** [review]
Just a few small comments below.  Also, can you delete the `package-lock.json` file?  `git rm package-lock.json`

That's generated by npm which we aren't using.

File: packages/web/src/components/Plan.tsx (hunk 2 above)
Comment (eccx400) [reply packages/web/src/components/Plan.tsx]: Changed

File: packages/web/src/components/Plan.tsx (hunk 4 above)
Comment (eccx400) [reply packages/web/src/components/Plan.tsx:133]: I tried that yesterday but I don't think the tutorial event appears on the localhost page; it seemed like the object wasn't getting pushed onto the array

File: packages/web/src/components/Plan.tsx (hunk 3 above)
Comment (eccx400) [reply packages/web/src/components/Plan.tsx]: hmm that's weird yea they're autoformatted on save even though I have prettier installed lemme look into this

File: packages/web/src/components/Plan.tsx (hunk 4 above)
Comment (eccx400) [reply packages/web/src/components/Plan.tsx:133]: I think in the newest commit I made a addTutorial function in the PlanHelpers.ts (modified addEvent) but only calls in the componentDidMount()

File: packages/web/src/components/Plan.tsx [hunk 5]
```diff
@@ -337,7 +364,7 @@ export class Plan extends React.Component<{}, State> {
                                 >
                                     <Button
                                         type="primary"
-                                        onClick={this.addEvent}
+                                        onClick={this.addTutorial}
```
Comment (drklee3) [inline packages/web/src/components/Plan.tsx]: This should add a new blank default event, not a tutorial one.

Tutorial event only shows on the *first* run when there's nothing saved in the user's browser.  Subsequent events added shouldn't be a tutorial event.

File: packages/web/src/components/Plan.tsx [hunk 6]
```diff
@@ -105,23 +79,69 @@ export class Plan extends React.Component<{}, State> {
...
+        console.log("The EventData description is " + theData.description);
+        console.log("The Elements in the Array is " + this.state.events);
+        console.log("The No. of Elements is " + this.state.events.length);
+        */
     }
 
     componentDidMount(): void {
         const restoredData = restoreData();
         const restoredPlan = restorePlan();
+        const tutorialEvent = new EventData(
```
Comment (drklee3) [inline packages/web/src/components/Plan.tsx:113]: Just use the `tutorialEvent` as the default event state when `restoredPlan` is undefined.  

`tutorialEvent` is being unused.  Your previous changes were more correct, just make a `tutorialEvent` here like you did, then set it as the event when `restoredPlan` is undefined on line 128-130.

File: packages/web/src/components/Plan.tsx (hunk 5 above)
Comment (eccx400) [reply packages/web/src/components/Plan.tsx]: oh my b I was just testing the function, I'll change it back

File: packages/web/src/components/Plan.tsx [hunk 7]
```diff
@@ -105,23 +106,69 @@ export class Plan extends React.Component<{}, State> {
...
+        //addEvent(arrData);
+        addTutorial(this.state.events);
+
+        console.log("The EventData is " + theData);
+        console.log("The EventData id is " + theData.id);
+        console.log("The EventData name is " + theData.name);
+        console.log("The EventData description is " + theData.description);
+        console.log("The Elements in the Array is " + this.state.events);
+        console.log("The No. of Elements is " + this.state.events.length);
+        */
```
Comment (drklee3) [inline packages/web/src/components/Plan.tsx]: Can be removed

File: packages/web/src/components/Plan.tsx [hunk 8]
```diff
@@ -105,23 +106,69 @@ export class Plan extends React.Component<{}, State> {
...
+
+        console.log("The EventData is " + tutorialEvent);
+        console.log("The EventData id is " + tutorialEvent.id);
+        console.log("The EventData name is " + tutorialEvent.name);
+        console.log(
+            "The EventData description is " + tutorialEvent.description
+        );
+        console.log("The Elements in the Array is " + this.state.events);
+        console.log("The No. of Elements is " + this.state.events.length);
```
Comment (drklee3) [inline packages/web/src/components/Plan.tsx]: Also can be removed

File: packages/web/src/components/Plan.tsx [hunk 9]
```diff
@@ -186,6 +234,12 @@ export class Plan extends React.Component<{}, State> {
         });
     };
 
+    addTutorial = (): void => {
+        this.setState({
+            events: addTutorial(this.state.events),
+        });
+    };
+
```
Comment (drklee3) [inline packages/web/src/components/Plan.tsx]: No longer needed

File: packages/web/src/itinerary/PlanHelpers.ts [hunk 10]
```diff
@@ -23,6 +23,18 @@ export function addEvent(events: EventData[]): EventData[] {
...
+        1,
+        "Tutorial Event",
+        new Date(),
+        new Date(),
+        "This is an example event! You can add or change descriptions and titles by clicking the edit button at the side. + search help when it's implemented, etc. etc."
+    );
+
+    return [newEvent];
+}
+
```
Comment (drklee3) [inline packages/web/src/itinerary/PlanHelpers.ts]: This too

File: packages/web/src/components/Plan.tsx [hunk 11]
```diff
@@ -105,23 +106,69 @@ export class Plan extends React.Component<{}, State> {
...
 
     componentDidMount(): void {
         const restoredData = restoreData();
         const restoredPlan = restorePlan();
+        const tutorialEvent = new EventData(
+            1,
+            "Tutorial Event",
+            new Date(),
+            new Date(),
+            "This is an example event! You can add or change descriptions and titles by clicking the edit button at the side. + search help when it's implemented, etc. etc."
```
Comment (drklee3) [inline packages/web/src/components/Plan.tsx]: Can you update the description?

**Review by drklee3 — COMMENTED** [review]
Seems like it works properly now, just need to remove some of the extra stuff and update the event description

File: packages/web/src/components/Plan.tsx (hunk 7 above)
Comment (eccx400) [reply packages/web/src/components/Plan.tsx]: Removed

File: packages/web/src/components/Plan.tsx (hunk 8 above)
Comment (eccx400) [reply packages/web/src/components/Plan.tsx]: Removed

File: packages/web/src/components/Plan.tsx (hunk 9 above)
Comment (eccx400) [reply packages/web/src/components/Plan.tsx]: Removed

File: packages/web/src/itinerary/PlanHelpers.ts (hunk 10 above)
Comment (eccx400) [reply packages/web/src/itinerary/PlanHelpers.ts]: Removed

**Review by drklee3 — COMMENTED** [review]
Last thing, whats the superhero submodule you just added?  Can you remove it?

**Comment by eccx400** [comment]
oh that was supposed to be for the mapping toggle on mobile, i thought i pushed it on another branch

**Review by drklee3 — APPROVED** [review]
LGTM! Nice work

**Pull request merged**